src/pages/patientView/PatientViewPage.tsx

```diff
@@ -544,6 +558,68 @@ export class PatientViewPageInner extends React.Component<
         return this.patientViewPageStore;
     }
 
+    @action.bound
+    private handleReturnToStudyView() {
+        const patientIdentifiers = this.pageStore.patientIdsInCohort.map(p => {
+            const patientIdParts = p.split(':');
+            return {
+                patientId: patientIdParts[1],
+                studyId: patientIdParts[0],
+            };
+        });
+        const queriedStudies: Set<string> = new Set(
+            patientIdentifiers.map(p => p.studyId)
+        );
+
+        const studyPage = window.open(
+            buildCBioPortalPageUrl(`study`, {
+                id: Array.from(queriedStudies).join(','),
+            }),
+            '_blank'
+        );
+        if (patientIdentifiers.length > 0) {
+            (studyPage as any).studyPageFilter = `filterJson=${JSON.stringify({
+                patientIdentifiers,
+            })}`;
+        }
+    }
+
+    @action.bound
+    private handleDeletePatient(deleteAtIndex: number) {
+        var newCohortIdList = toJS(this.pageStore.patientIdsInCohort);
+        newCohortIdList.splice(deleteAtIndex, 1);
+        let currentIndex =
+            deleteAtIndex > newCohortIdList.length - 1
+                ? deleteAtIndex - 1
+                : deleteAtIndex;
+
+        let navCaseIds = newCohortIdList.map(p => {
+            const patientIdParts = p.split(':');
+            return {
+                patientId: patientIdParts[1],
+                studyId: patientIdParts[0],
+            };
+        });
+        const url = getPatientViewUrl(
+            navCaseIds[currentIndex].studyId,
+            navCaseIds[currentIndex].patientId,
+            navCaseIds
+        );
+
+        // update hash if url lenght is less than max length
```

Comment:
hey, can you add to this:  
Because of url length limits, we can only maintain the list in the url hash for small sets of ids.